api/rest/restapi.go

```diff
@@ -367,6 +367,18 @@ func (api *API) routes() []route {
 			"/pins/{hash}",
 			api.unpinHandler,
 		},
+		{
+			"PathPin",
+			"POST",
+			"/pins/{keyType:ipfs|ipns}/{path:.*}",
```

Comment:
we might need to handle `/ipld/...` paths too. I did not know they were a thing but just saw them mentioned. Can you find out how they differ from `/ipfs/` paths and see if they need some special treatment?

Reply:
Trying out `ipfs resolve` with ipld addresses and going through github.com/ipfs/go-path, it doesn'r seem like it would need any special treatment. It seems that `/ipld` are using key format as `/ipfs`, they seem quite interchangable. But to be sure of this, I am yet to go through [ipfs refs code](https://github.com/ipfs/go-ipfs/blob/master/core/commands/refs.go)